src/main/java/bio/terra/workspace/service/datareference/utils/DataReferenceValidationUtils.java

```diff
@@ -1,56 +1,67 @@
 package bio.terra.workspace.service.datareference.utils;
 
-import static bio.terra.workspace.generated.model.ReferenceTypeEnum.DATA_REPO_SNAPSHOT;
-
-import bio.terra.workspace.generated.model.DataRepoSnapshot;
-import bio.terra.workspace.generated.model.ReferenceTypeEnum;
 import bio.terra.workspace.service.datareference.exception.InvalidDataReferenceException;
+import bio.terra.workspace.service.datareference.model.DataReferenceType;
+import bio.terra.workspace.service.datareference.model.ReferenceObject;
+import bio.terra.workspace.service.datareference.model.SnapshotReference;
 import bio.terra.workspace.service.datarepo.DataRepoService;
 import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
 import java.util.regex.Pattern;
 import org.apache.commons.lang3.StringUtils;
 import org.springframework.stereotype.Component;
 
+/** A collection of validation functions for data references. */
 @Component
 public class DataReferenceValidationUtils {
 
   private DataRepoService dataRepoService;
-  private final Pattern nameValidationPattern = Pattern.compile("^[a-zA-Z0-9][_a-zA-Z0-9]{0,62}$");
 
-  public DataReferenceValidationUtils(DataRepoService dataRepoService) {
-    this.dataRepoService = dataRepoService;
-  }
+  /**
+   * Names must be 1-63 characters long, and may consist of alphanumeric characters and underscores
+   * (but may not start with an underscore). These restrictions match TDR snapshot name
+   * restrictions.
+   */
+  public static final Pattern NAME_VALIDATION_PATTERN =
```

Comment:
Why do WM data reference names need to match snapshot name restrictions?

Reply:
Partially because we expect WM references to snapshots to frequently use the underlying snapshot's name (though this isn't required or enforced), partially because this seemed like a reasonable standard for human generated + readable names.

Reply:
Can we add that info to the comment? As-is, it sounds like the restrictions must match the TDR snapshot name restrictions.